metron-platform/metron-common/src/main/config/zookeeper/global.json

```diff
@@ -6,5 +6,16 @@
   "update.hbase.table": "metron_update",
   "update.hbase.cf": "t",
   "es.client.settings": {
-  }
+  },
+  "parser.groups": [
```

Comment:
What is this option in global config? I don't recall seeing this before, and it's not mentioned anywhere from what I can tell, e.g. https://github.com/apache/metron/tree/master/metron-platform/metron-common#global-configuration.

Reply:
Parser groups are stored in the global config.  There are REST endpoints for managing parser groups, a user would not directly edit this in the global config.  I can add it to that README but I'm not sure it's useful to a user.  I'm happy to add more documentation wherever it makes sense.

Reply:
why would we store them there and not in zookeeper?

Reply:
> why would we store them there and not in zookeeper?

@ottobackwards not sure I completely understand the Q here - global.json is stored in ZK. Is your question why this is stored in global.json?

Part of the reason I asked about the docs is because I don't have a good sense of the implications to storing this here. This is the original PR - https://github.com/apache/metron/pull/1346. @merrimanr what happens when I start a parser group from the CLI, will global.json be out of sync for the UI then?

Reply:
There is no change to starting a parser group from the CLI.  You would still pass them in as a comma-separate list of parsers if you wanted to group parsers together.  Same goes for Ambari.  This setting is only used if parser topologies are managed in the Management UI.  The REST service handles converting parser groups to the correct CLI arguments and Storm job names.

Reply:
Yes, I'm asking why they are in the global configuration and not just a new ZK key/node

Reply:
I think the question here then becomes why this config change is made in this PR and not part of the UI PR that supports aggregation? If this is literally a "the 2 PR's are tightly coupled and should be merged together" that's probably fine. But I'm currently confused as to what the grouping change is doing here since there doesn't seem to be an existing endpoint/feature that this is impacting.

Reply:
I'm happy to resubmit this against the feature branch if that makes reviewing easier.

Reply:
Just to be clear, I think the overall PR is fine, it's just the recent parser group change to global.json that looks like it belongs against the feature branch. Does that work?

Reply:
Yes no problem.

Reply:
> Parser groups are stored in the global config. There are REST endpoints for managing parser groups, a user would not directly edit this in the global config. I can add it to that README but I'm not sure it's useful to a user. I'm happy to add more documentation wherever it makes sense.

Can you add documentation for this feature? It looks like it was missed in the original PR.

Reply:
What documentation do you want me to add?  Do you want a dedicated section on parser groups somewhere?  Just add the setting to the global config?

Reply:
Something to explain what it is if I've never seen this config option before, but am familiar with global config. From what you said earlier, this is purely a sharing mechanism, not something a user should ever modify. Some of this may be defined implicitly from the parser aggregation feature, mostly concerned with how this additional feature functions in concert with it.

This is just off the top of my head skimming through, so I imagine you'll have a lot more to add to it as the implementer. Some of this may already be documented out there somewhere, just want to make sure we put a bow on it.

**User-level concerns**

1. If a user does modify it manually, whether intentionally or by accident, what will happen?
2. What processes manage this config? Will Ambari or the config UI manage or overwrite these values?
3. If a user adds a group manually, will the values remain, or get overwritten/removed? (e.g. what's the interaction with our global config JSON Patching?)
4. How is a group different from a parser aggregation as defined in the rest of our documentation?

**Developer concerns**

1. Same as above, plus
2. What is the unique ID for a group, is there one?
3. What if I group the same sensors in multiple groups?
4. What are each of the fields and how are they used? i.e. If I go modify this feature in the UI, what is my API and what are the guarantees it provides?
5. What fields are required vs optional? Related to the "guarantees" in 4. Important for null checks, type checks, etc. etc.

Reply:
I spent some time trying to document this and I realized the documentation doesn't make much sense because the UI component hasn't been done yet.  I find myself guessing how it is going to work because I'm not 100% clear on how the UI will function and I don't have anything to reference because that work is still being submitted.  Almost all of the requests in your list involve the UI in some way.  Parser group and parser aggregation are synonymous so that's something else we'll want to fix. 

I think I may put this on hold until the UI part is further along unless you would accept that work as a follow on Jira that blocks the acceptance of this feature branch.  The benefit of committing this now is it will resolve a bug that was found earlier.